Assets/Mirage/ObjectPolling/PooledObject.cs

```diff
@@ -0,0 +1,262 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Mirage.Runtime
+{
+    public struct PooledObject<T>
+    {
+        public T Object;
+        public NetworkIdentity NetId;
+    }
+
+    internal struct ObjectPooling<T> where T : Object
```

Comment:
does this need to be a struct?

Reply:
Yes it does because of constantly call new () when i pool it back. Doing that on a class is huge performance and garbage colllection no no.